app/controllers/queue_items_controller.rb

```diff
@@ -22,4 +35,16 @@ def destroy
   def queue_video(video)
     QueueItem.create(video: video, user: current_user, position:  current_user.new_queue_item_position) unless current_user.queued_video?(video)
   end
+
+  def update_queue_items
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |queue_item_data|
+        queue_item = QueueItem.find(queue_item_data["id"])
+        if queue_item.user == current_user
+          queue_item.update_attributes!(position: queue_item_data["position"]) 
+          queue_item.rating=(queue_item_data["rating"]) if queue_item_data["rating"]
```

Comment:
HOMEWORK: Refactor lines 44 and 45, `update_attributes!` to `update!` and `rating=` will be called by `update`. Check out the solution.